mwdb/web/src/components/ShowObject/Views/RelationsBox.js

```diff
@@ -0,0 +1,130 @@
+                    </span>
+                </td>
+                <td>
+                    <TagList tags={parent.tags}/>
+                </td>
+            </tr>
+        );
+
+    const children = (props.children || [])
+        .map((child, index, array) =>
+            <tr key={`child-${child.id}`} className="flickerable">
+                <th>child</th>
+                <td>
+                    <span>
+                        <ObjectLink {...child }
+                                    diffWith={child.type === "text_blob" && (array[index + 1] || {}).id}
+                                    inline/>
+                    </span>
+                    <span className="ml-2">
+                        <ActionCopyToClipboard text={child.id} tooltipMessage="Copy sha256 to clipboard"/>
+                    </span>
+                </td>
+                <td>
+                    <TagList tags={child.tags}/>
+                </td>
+            </tr>
+        );
+    return (
+        <div className="card card-default">
+            <div className="card-header">
+                { props.header || "Relations" }
+                <Link to="#" className="float-right"
+                      onClick={(ev) => {
+                          ev.preventDefault();
+                          setAttributeAddModalOpen(true);
+                          setModalError("");
+                      }}>
+                        <FontAwesomeIcon icon="plus" pull="left" size="1x"/>
+                        Add
+                </Link>
+            </div>
+            {
+                parents.length + children.length > 0 ?
+                <table className="table table-striped table-bordered table-hover" id="rel_table">
+                    <tbody id="rel_body">
+                    { parents }
+                    { children }
+                    </tbody>
+                </table> : <div className="card-body text-muted">No relations to display</div>
+            }
+            <RelationsAddModal isOpen={isAttributeAddModalOpen}
+                               error={modalError}
+                               onSubmit={addObjectRelations}
+                               onError={setModalError}
+                               onRequestModalClose={() => setAttributeAddModalOpen(false)}
+            />
+        </div>
+    );
+}
+
+export default function MultiRelationsBox(props) {
+    const context = useContext(ObjectContext);
+
+    let TypedRelationsBox = (props) => {
+        let filterByType = function(arr) { return arr.filter(e => e.type === props.type) }
+        let filteredElements = {
+            parents: filterByType(props.parents),
+            children: filterByType(props.children)
+        }
+        if(filteredElements.parents.length + filteredElements.children.length > 0)
+            return <RelationsBox {...filteredElements} header={props.header} id={context.object.id}/>
+        else
+            return <div/>
```

Comment:
It's a bit overengineered IMO (I know it's copied from the original code)
```suggestion
        const parents = props.parents.filter(e => e.type === props.type);
        const children = props.children.filter(e => e.type === props.type);
        if(parents.length + children.length > 0)
            return <RelationsBox header={props.header} id={context.object.id} {...{parents, children}} />
        else
            return <div/>
```